hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -587,15 +601,15 @@ void logSlowRpcCalls(String methodName, Call call,
     final double threeSigma = rpcMetrics.getProcessingMean() +
         (rpcMetrics.getProcessingStdDev() * deviation);
 
-    long processingTime =
-            details.get(Timing.PROCESSING, rpcMetrics.getMetricsTimeUnit());
+    final TimeUnit metricsTimeUnit = rpcMetrics.getMetricsTimeUnit();
+    long processingTime = details.get(Timing.PROCESSING, metricsTimeUnit);
+    long logSlowRPCThresholdTime =
+        metricsTimeUnit.convert(getLogSlowRPCThresholdMs(), TimeUnit.MILLISECONDS);
```

Comment:
There is no need to covert this threshold every time. We can convert this threshold when it is initialized by this timeunit, because this timeunit will not be changed.